### ISSUE TITLE
v13.0.0 - Can't find stylesheet to import

### ISSUE THREAD

**Issue by sheaivey** [HealthCatalyst/Fabric.Cashmere#1905]
**Summary**

In the latest cashmere v13.0.0 release `package.json` added the `exports` key which blocks access to anything not explicitly exported from the module. Because `hcicons` and `scss/cashmere` are not exported I am getting an error that they cannot be found.
```scss
/* file: ./src/index.scss */

@import '~@healthcatalyst/cashmere/scss/cashmere';
```

```
Compiled with problems:

Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: Can't find stylesheet to import.
   ╷
10 │ @import '~@healthcatalyst/cashmere/scss/cashmere';
   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   ╵
  src/index.scss 10:9  root stylesheet
```

**Reproduction**

Steps to reproduce:

1. Create a clean react project.
2. `npx create-react-app my-app`
3. `cd my-app`
4. `npm install @healthcatalyst/cashmere sass`
5. `npm run start`
7. Rename `./src/index.css` to `./src/index.scss` and update the import in `./src/index.js` to the new file name.
6. Verify the app runs on localhost:3000 (you should see a react logo spinning)
8. in `./src/index.scss` add the line `@import '~@healthcatalyst/cashmere/scss/cashmere';`
9. You should now have an error saying it could not find the cashmere file.

**Additional context**
I believe the `exports` key is specific to webpack loaders.
https://webpack.js.org/guides/package-exports/

The fix is to add entries under `exports` for anything you want people to be able to import from the cashmere package.

An example solution might look something like this...
```json
"exports": {
/* start of new entries */
    "./cashmere": {
      "default": "./cashmere.css"
    },
    "./hcicons/hcicons": {
      "default": "./hcicons/hcicons.scss"
    },
    "./scss/cashmere": {
      "default": "./scss/cashmere.scss"
    },
/* end of new entries */
    "./package.json": {
      "default": "./package.json"
    },
    ".": {
      "types": "./healthcatalyst-cashmere.d.ts",
      "esm2020": "./esm2020/healthcatalyst-cashmere.mjs",
      "es2020": "./fesm2020/healthcatalyst-cashmere.mjs",
      "es2015": "./fesm2015/healthcatalyst-cashmere.mjs",
      "node": "./fesm2015/healthcatalyst-cashmere.mjs",
      "default": "./fesm2020/healthcatalyst-cashmere.mjs"
    }
  },
```

**Comment by corykon** [comment]
@andrew-frueh I think @sheaivey found the source of our import issues. the tilde syntax is deprecated, so it's still good we're planning to change that guidance, but I'd bet if we update our exports in package.json as Shea suggests, this syntax might work everywhere: `@import '@healthcatalyst/cashmere/scss/cashmere'`

**Comment by sheaivey** [comment]
Glad to help! I can confirm that removing the `~` and switching `@import` to `@use` still resolves the newly exported items.

I was trying to follow the angular cashmere build config to see how those exports could be automatically added but my experience there is lacking. 

From the docs it appears we could manually add `exports` to `package.json` and the build will merge any auto generated exports with that. 
https://angular.io/guide/creating-libraries#managing-assets-in-a-library

**Comment by corykon** [comment]
Thanks a ton for pointing this stuff out @sheaivey.

I have a PR up now, and I sent you an invite to be a collaborator on this repo so you can be added as a PR reviewer.

Led me to some studying on webpack imports/exports and the way the new sass modules work that I should've done awhile back. We'll probably want to update our codebase and our guidance on @import vs @use here sometime soon.

https://webpack.js.org/guides/package-exports/
https://stefaniefluin.medium.com/the-new-sass-module-system-out-with-import-in-with-use-e1bd8ba032d0#:~:text=The%20new%20%40use%20rule%20is,reference%20it%20in%20your%20project.

**Comment by andrew-frueh** [comment]
Closed in #1907
